RHDEVDOCS-4940-firstPR: Covers CVE update for GitOps 1.7.1, 1.6.4, and 1.5.9

shipsing opened the pull request:
**NOTE**: From GitOps 1.7.0 onwards, we will be documenting security issues in a new advisory format as part of errata updates. 

**Aligned team**: Dev Tools

**Purpose**: To resolve the following issues: 
https://issues.redhat.com/browse/RHDEVDOCS-4907
https://issues.redhat.com/browse/RHDEVDOCS-4940

**OCP version this PR applies to**:  enterprise 4.9 and later

**Link to docs preview**:  https://55464--docspreview.netlify.app/openshift-enterprise/latest/cicd/gitops/gitops-release-notes.html#gitops-release-notes-1-7-1_gitops-release-notes

**Change Management**
Need an ACK from Eng: @iam-veeramalla 
Need an ACK from PM: @harrietgrace @wtam2018 
Need an ACK from Product experience: @RickJWagner 
Need an ACK from QE: @varshab1210 
Need an ACK from DPM: @pmkovar 
Need an ACK from CS: @Preeticp 

Peer review: @gabriel-rh

shipsing commented:
@iam-veeramalla @harrietgrace @wtam2018 @RickJWagner @varshab1210 @pmkovar @Preeticp, Could you please help with review of this PR and share your feedback? Thanks

harrietgrace commented on modules/gitops-release-notes-1-7-1.adoc:
```diff
@@ -0,0 +1,25 @@
+
+If you have installed the {gitops-title} Operator, run the following command to view the container images in this release:
+
+[source,terminal]
```
Will this command change with each version?

shipsing replied:
@iam-veeramalla, Could you please provide your inputs here? Also, I added this command to make it look consistent with the format we followed in the OCP RN here: https://docs.openshift.com/container-platform/4.11/release_notes/ocp-4-11-release-notes.html#ocp-4-11-2

iam-veeramalla replied:
it will be constant.

harrietgrace commented on modules/gitops-release-notes-1-7-1.adoc:
```diff
@@ -0,0 +1,25 @@
+{gitops-title} 1.7.1 is now available on {product-title} 4.8, 4.9, 4.10, 4.11, and 4.12.
+
+[id="errata-updates-1-7-1_{context}"]
+== Errata updates
```
This release only has security fixes, right? When we do a release that has more than just the Errata section, are we able to make sure the Errata goes last? 😊

shipsing replied:
yes, @harrietgrace. Organization and placement of our content is in our hands. Since we did not have anything else for the GitOps 1.7.1 release, other than these security fixes, we had only one section of errata updates. In future releases, if we have other sections about new features, bug fixes, known issues, we would make sure to keep any errata updates in the end.

harrietgrace commented on modules/gitops-release-notes-1-7-1.adoc:
```diff
@@ -0,0 +1,25 @@
+
+Issued: 2023-01-25
+
+{gitops-title} release 1.7.1, which includes security updates, is now available. The list of security fixes that are included in the update is documented in the link:https://access.redhat.com/errata/RHSA-2023:0467[RHSA-2023:0467] advisory. 
```
Looking at the OpenShift version of these, it seems like this is a standard block of copy. Do we have any leeway to adjust it, or do we need to keep it exactly like this? 

If we're able to change it each time we add this Errata block - what do you think about removing the first sentence? Given that we're putting the Errata in with each version's RNs it feels repetitive to say that the release is now available. We could add the "which includes security updates" part to Line 9 if we want it to be mentioned somewhere?

shipsing replied:
@Preeticp, need your inputs here. 
The first sentence suggests that a newer version (1.7.1) is available, though the same information is mentioned a  few lines above as well. So, this makes the content a bit redundant. I guess, we can remove the first sentence and start the para from the second sentence `The list of security fixes that are included in the update is documented...`  If you look at the preview, you can guage it better: https://55464--docspreview.netlify.app/openshift-enterprise/latest/cicd/gitops/gitops-release-notes.html#gitops-release-notes-1-7-1_gitops-release-notes

Preeticp replied:
@shipsing +1, tiny tweak: The list of security fixes that are included in this release is documented in the [RHSA-2023:0467](https://access.redhat.com/errata/RHSA-2023:0467) advisory.

harrietgrace reviewed (commented):
I'm not very familiar with the docs standards here, so please forgive my noob questions 😄

RickJWagner commented:
/label px-approved

shipsing commented:
Copied the content from https://github.com/openshift/openshift-docs/pull/55529 into this PR to have a single PR to track security fixes for GitOps 1.5.9, 1.6.4, and 1.7.1. This PR will be cherrypicked to 4.8 and later. Thanks

cc @iam-veeramalla @harrietgrace @wtam2018 @RickJWagner @varshab1210 @pmkovar @Preeticp

iam-veeramalla commented on modules/gitops-release-notes-1-7-1.adoc:
```diff
@@ -0,0 +1,25 @@
+
+Issued: 2023-01-25
+
+The list of security fixes that are included in this release is documented in the link:https://access.redhat.com/errata/RHSA-2023:0467[RHSA-2023:0467] advisory. 
```
Why is it different for 1.7.1

I see for 1.5.9 and 1.6.4, you mentioned the bugzilla issues. I think it is better to do the same thing here as well.

shipsing replied:
@iam-veeramalla I have added a note in the PR description. From GitOps 1.7.0 onwards, we are going to use this new advisory fomat for all security fixes. To avoid any consistency issue with previous releases (1.6.4 and 1.5.9), we have used the same language that was there in the already published RN. The intent is to segregate security fixes from regular bug fixes.

Preeticp replied:
@Srivaralakshmi please note ^

shipsing commented:
@harrietgrace I have incorporated your suggestions and provided rationale where required. Please verify the same and confirm.

harrietgrace reviewed (approved):
Looks great, thanks @shipsing!

wtam2018 commented:
/lgtm

varshab1210 commented:
/lgtm

shipsing commented:
I have included a few more updates into this PR to avoid another cycle of change management:
https://issues.redhat.com/browse/RHDEVDOCS-4907: As part of this update, i incorporated the gaps identified for z stream release notes, basically this comment: https://issues.redhat.com/secure/EditComment!default.jspa?id=15043528&commentId=21626794

https://issues.redhat.com/browse/RHDEVDOCS-4940: Besides covering CVE fixes, I have also added a new feature entry in GitOps 1.7.0 RN for this issue: https://issues.redhat.com/browse/GITOPS-2340.

@iam-veeramalla @harrietgrace @wtam2018 @RickJWagner @varshab1210 @pmkovar @Preeticp, please review the updated PR and share your feedback. These are the new updates: https://github.com/openshift/openshift-docs/compare/d8a39028732a9a27b9480d6f178f26ea37404252..4566920648bf9c2c23b05407467ee13aa020f60b

varshab1210 commented:
Looks good to me. Thanks @shipsing!
/lgtm

Preeticp commented:
LGTM from the CS side

harrietgrace commented:
lgtm from PM 😊

pmkovar commented:
LGTM

gabriel-rh commented on modules/gitops-release-notes-1-5-9.adoc:
"for audiences who might not be intended to access"

This sounds a bit off - maybe "would accept tokens allowing unintended access to the cluster" but any such simplification would be good.  I would definitely change "audiences" to plain "users".

Srivaralakshmi commented on cicd/gitops/gitops-release-notes.adoc:
```diff
@@ -23,14 +23,20 @@ include::modules/go-compatibility-and-support-matrix.adoc[leveloffset=+1]
 
 include::modules/gitops-release-notes-1-6-0.adoc[leveloffset=+1]
 
+include::modules/gitops-release-notes-1-5-9.adoc[leveloffset=+1]
```
**Observation:** [GITOPS-1756](https://issues.redhat.com/browse/GITOPS-1756) is already documented in 4.8 OCP as Fixed Issues. 

**Suggestion:** 

- So no need of repeating it in RN 1.4.2 and 1.3.3. 
- Also, add the RN module 1.2.2 in OCP version 4.9 and later.

shipsing replied:
Incorporated this suggestion @Srivaralakshmi. Please verify and confirm. Thanks

gabriel-rh commented on modules/gitops-release-notes-1-6-2.adoc:
```diff
@@ -9,6 +9,11 @@
+[id="new-features-1-6-2_{context}"]
+== New features
+
+* This release removes the `DISABLE_DEX` environment variable from the `openshift-gitops-operator` CSV file. As a result, this environment variable is no longer set when you do a fresh installation of {gitops-title}. link:https://issues.redhat.com/browse/GITOPS-2360[GITOPS-2360]
```
```suggestion
* This release removes the `DISABLE_DEX` environment variable from the `openshift-gitops-operator` CSV file. As a result, this environment variable is no longer set when you perform a fresh installation of {gitops-title}. link:https://issues.redhat.com/browse/GITOPS-2360[GITOPS-2360]
```

Srivaralakshmi reviewed (changes requested):
@shipsing Thanks for the good work! Have put some comments. PTAL.

gabriel-rh commented on modules/gitops-release-notes-1-7-0.adoc:
```diff
@@ -45,6 +45,12 @@ In future releases, there is a possibility to deprecate the old method of custom
 include::snippets/technology-preview.adoc[]
 
+* With this update, Argo CD supports the Server-Side Apply feature, which helps users to perform the following tasks:
+** Manage large resources which are too big to fit in the 262144 bytes allowed annotation size.
```
"Manage large resources which are too big for the allowed annotation size of 262144 bytes." maybe?

gabriel-rh reviewed (commented):
This looks good to me - some very minor suggestions.
Good work, as usual!

shipsing commented:
@gabriel-rh Incorporated your suggestions. Please verify and confirm. Thanks for the review.

Srivaralakshmi reviewed (approved):
@shipsing Thanks for the changes. LGTM.

Srivaralakshmi commented:
/lgtm

gabriel-rh reviewed (approved):
/lgtm

rolfedh commented:
/cherry-pick enterprise-4.9

rolfedh commented:
/cherry-pick enterprise-4.10

rolfedh commented:
/cherry-pick enterprise-4.11

rolfedh commented:
/cherry-pick enterprise-4.12

rolfedh commented:
/cherry-pick enterprise-4.13

RickJWagner commented:
Looks good.  Thank you, @shipsing!
